### ISSUE TITLE
Increase max zoom to 24

### ISSUE THREAD

**Issue by jywarren** [publiclab/Leaflet.DistortableImage#473]
We can no longer zoom in very close; it used to allow up to zoom 24, but we're now limited to much lower (18?) -- but here is a post about the high-res we used to have:

https://publiclab.org/notes/warren/01-19-2018/mapknitter-update-very-high-resolution-mapping (It was added in this PR: https://github.com/publiclab/mapknitter/pull/243/files#diff-cd467f0b470669270454ce0ae6227a90R492-R493)

I think this was changed when [GoogleMutant was moved into the Leaflet.DistortableImage library](https://github.com/publiclab/mapknitter/pull/959/files#diff-cd467f0b470669270454ce0ae6227a90R615)! 

But the fix is pretty simple! Just these lines:

https://github.com/publiclab/Leaflet.DistortableImage/blob/6e4b078eb83ddefa191a726be8a2671640897cba/src/mapmixins/MapMixins.js#L11-L12

Have to be changed to match these (see docs at https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant):

```js
    var googleMutant = L.gridLayer.googleMutant({
      type: 'satellite', // valid values are 'roadmap', 'satellite', 'terrain' and 'hybrid'
      maxZoom: 24,
      maxNativeZoom: 20,
      opacity:0.5
    }).addTo(this._map);
```

We'd love help with this and it would be a great GCI task! cc @sashadev-sky

**Comment by sashadev-sky** [comment]
@jywarren I just doublechecked and the reason the maxZoom is set to 21 is because beyond that the tiles disappear and you just have a blank gray background. 

I'll look into how you got it to 24 before. I don't remember what I didn't like about the google mutant plugin functionality, but I'll give it a try again to see

**Comment by sashadev-sky** [comment]
Just looked at the plugin again. It needs an API key so I think we should just update MK to add the old google mutant layer from the plugin instead of making any changes upstream. 

lmk and if so, I will close this and make a PR in MK because its high-priority / we have a lot of `gci-candidates` in this repo that still haven't been uploaded to dashboard

**Comment by jywarren** [comment]
OK, well if you can just do it via MapKnitter that's ok, but i think it's a really common use case for distorting maps in LDI. 

There is a special `maxNativeZoom: 20,` option I set using  https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant which allows for zooming past the highest zoom level available; it stretches the tiles to make this possible. Pretty cool!

**Comment by jywarren** [comment]
Ah, you moved off of  https://gitlab.com/IvanSanchez/Leaflet.GridLayer.GoogleMutant and that's why we're not getting this anymore? Can you try using `maxNativeZoom` - I actually believe it's a Leaflet feature, not a Google Mutant feature, so I think it'll work!

Yes, take a look here: https://leafletjs.com/reference-1.4.0.html#gridlayer-maxnativezoom

So this ought to be easy! Thank you!!

**Comment by sashadev-sky** [comment]
@jywarren we will get errors for any zoom beyond 21 even with `maxNativeZoom` set. Its the url for the tile layers were using. So for these tiles I think we have to limit the zoom. But the LDI API is made to use with any tiles, if you just skip the `addGoogleMutant` call and add your own. So it will be easy to update upstream. I will add a PR.

downstream I feel like its ok because people can use whatever tiles they'd like anyway? It mentions this in README: https://github.com/publiclab/Leaflet.DistortableImage#single-image-interface

> Note: map.addGoogleMutant() is a convenience function for adding our recommended layer to the map. If you want a different baselayer, skip this line and add your preferred setup instead.

**Comment by jywarren** [comment]
Hm, that's weird, it was definitely working before, what has changed since
that blog post was written?

**Comment by sashadev-sky** [comment]
@jywarren sorry I take that back. I was just not setting the maxNativeZoom on the labels layer as well. I will upload a PR here.

Also slightly unrelated q while were on this topic: does it make a difference if we're using http or https here?
https://github.com/publiclab/Leaflet.DistortableImage/blob/67bb5dbe69bfeaf31e16aabc08e4a36b476a1715/src/mapmixins/MapMixins.js#L7

**Comment by jywarren** [comment]
I think i just fixed this in #486, want to take a look?

I'm not sure about the url -- but often we can use `//example.com` and it will default in the browser to http/https depending on what the overall page is using -- if that's helpful? But i think it may not be needed for us this specific bug?

Thanks!!!
